src/interfaces/node.h

```diff
@@ -188,6 +188,9 @@ class Node
     //! Return interfaces for accessing wallets (if any).
     virtual std::vector<std::unique_ptr<Wallet>> getWallets() = 0;
 
+    //! Attempts to load a wallet from file or directory.
+    virtual std::unique_ptr<Wallet> loadWallet(const std::string& name, std::string& error, std::string& warning) = 0;
```

Comment:
Currently `BitcoinGUI::setCurrentWallet` is only called when a wallet is opened with the GUI not after `handleLoadWallet` notification. For that reason it's returning the wallet interface. Beside, I was under the impression that it's valid to have multiple interface instances to the same underlying wallet/node/etc.

Reply:
> I was under the impression that it's valid to have multiple interface instances to the same underlying wallet/node/etc.

It will work, but I assume will cause unnecessary complexity and code duplication, because now there will be two different ways for the GUI to be updated when the wallet is loaded instead of just one. I think ideally this PR would add a new load command but reuse existing mechanisms for attaching new wallets to the GUI.

Whatever approach you take, I think should be a comment describing how this interacts with the load wallet notification.

Reply:
> now there will be two different ways for the GUI to be updated when the wallet is loaded instead of just one

I think the current approach makes sense (as in `wallet = loadwallet(...)`) in the same way Qt has `QFileDialog::getOpenFileName` — the implementation is not ideal but some refactors are needed to make this better which I intend to work on.

I'll add comments in the next push.

Reply:
Added a small comment regarding the notification.